metals/src/main/scala/scala/meta/internal/implementation/SuperMethodProvider.scala

```diff
@@ -0,0 +1,211 @@
+    } else {
+      None
+    }
+  }
+
+  private def getSuperClasses(
+      si: SymbolInformation,
+      findSymbol: String => Option[SymbolInformation],
+      skip: scala.collection.mutable.Set[SymbolInformation]
+  ): List[(SymbolInformation, Option[TextDocumentWithPath])] = {
+    if (skip.exists(_.symbol == si.symbol)) {
+      List()
+    } else {
+      skip += si
+      si.signature match {
+        case clz: ClassSignature =>
+          val parents = clz.parents
+            .collect { case x: TypeRef => x }
+            .filterNot(si =>
+              Set(
+                "scala/AnyRef#",
+                "scala/Serializable#",
+                "java/io/Serializable#",
+                "scala/AnyVal#"
+              ).contains(si.symbol)
+            )
+            .filterNot(p => skip.exists(_.symbol == p.symbol))
+          val mainParents = parents.headOption
+            .flatMap(p => findSymbol(p.symbol))
+            .map(si => getSuperClasses(si, findSymbol, skip))
+            .toList
+            .flatten
+          val withParents = if (parents.nonEmpty) {
+            parents.tail.reverse.flatMap { linParent =>
+              findSymbol(linParent.symbol)
+                .map(ssss => getSuperClasses(ssss, findSymbol, skip))
+                .toIterable
+                .flatten
+            }.toList
+          } else {
+            List.empty
+          }
+          val expParents = withParents ++ mainParents
+          val semDB = findSemanticDbWithPathForSymbol(si.symbol)
+          (si, semDB) +: expParents
+        case sig: TypeSignature =>
+          findSymbol(sig.lowerBound.asInstanceOf[TypeRef].symbol)
+            .filterNot(s => skip.exists(_.symbol == s.symbol))
+            .map(getSuperClasses(_, findSymbol, skip))
+            .getOrElse(List())
+        case _ =>
+          List()
+      }
+    }
+  }
+
+  private def calculateClassSuperHierarchyWithCache(
+      classSymbolInformation: SymbolInformation,
+      cache: LensGoSuperCache,
+      findSymbol: String => Option[SymbolInformation]
+  ): List[(SymbolInformation, Option[TextDocumentWithPath])] = {
+    cache.get(classSymbolInformation.symbol) match {
+      case Some(value) => value
+      case None =>
+        val value =
+          calculateClassSuperHierarchy(classSymbolInformation, findSymbol)
+        cache(classSymbolInformation.symbol) = value
+        value
+    }
+
+  }
+
+  private def calculateClassSuperHierarchy(
+      classSymbolInformation: SymbolInformation,
+      findSymbol: String => Option[SymbolInformation]
+  ): List[(SymbolInformation, Option[TextDocumentWithPath])] = {
+    val result = getSuperClasses(
+      classSymbolInformation,
+      findSymbol,
+      scala.collection.mutable.Set[SymbolInformation]()
+    )
+    println(s"${classSymbolInformation.symbol} ==> ${result.map(_._1.symbol)}")
+    result
+  }
+
+  def findSuperForMethodOrFieldChecked(
```

Comment:
You should use `MethodImplementation.findParentSymbol`, no need to reimplement it all.

Reply:
I don't think I could use that as `MethodImplementation.findParentSymbol` doesn't care about linearization and does not necessarily finds next parent in linearized hierarchy. 
We can discuss that if you are convinced it is usable in this context.